harper-core/src/linting/closed_compounds.rs

```diff
@@ -0,0 +1,342 @@
+        }
+
+        impl PatternLinter for $name {
+            fn pattern(&self) -> &dyn Pattern {
+                self.pattern.as_ref()
+            }
+
+            fn match_to_lint(&self, matched_tokens: &[Token], source: &[char]) -> Lint {
+                let span = matched_tokens.span().unwrap();
+                let orig_chars = span.get_content(source);
+
+                Lint {
+                    span,
+                    lint_kind: LintKind::WordChoice,
+                    suggestions: vec![Suggestion::replace_with_match_case(
+                        $correct.chars().collect(),
+                        orig_chars,
+                    )],
+                    message: format!("Did you mean the closed compound `{}`?", $correct),
+                    ..Default::default()
+                }
+            }
+
+            fn description(&self) -> &'static str {
+                concat!(
+                    "Looks for incorrect spacing inside the closed compound `",
+                    $correct,
+                    "`."
+                )
+            }
+        }
+    };
+}
+
+create_closed_compound_linter!(Itself, "it self", "itself");
+create_closed_compound_linter!(Tonight, "to night", "tonight");
+create_closed_compound_linter!(Myself, "my self", "myself");
+create_closed_compound_linter!(Therefore, "there fore", "therefore");
+create_closed_compound_linter!(Misunderstand, "miss understand", "misunderstand");
+create_closed_compound_linter!(Misunderstood, "miss understood", "misunderstood");
+create_closed_compound_linter!(Misuse, "miss use", "misuse");
+create_closed_compound_linter!(Misused, "miss used", "misused");
+create_closed_compound_linter!(Postpone, "post pone", "postpone");
+create_closed_compound_linter!(Worldwide, "world wide", "worldwide");
+create_closed_compound_linter!(Overall, "over all", "overall");
+create_closed_compound_linter!(Likewise, "like wise", "likewise");
+create_closed_compound_linter!(However, "how ever", "however");
+create_closed_compound_linter!(Altogether, "all together", "altogether");
+create_closed_compound_linter!(Asleep, "a sleep", "asleep");
+create_closed_compound_linter!(Upset, "up set", "upset");
+create_closed_compound_linter!(Intact, "in tact", "intact");
+create_closed_compound_linter!(Somehow, "some how", "somehow");
+create_closed_compound_linter!(Proofread, "proof read", "proofread");
+create_closed_compound_linter!(Nobody, "no body", "nobody");
+create_closed_compound_linter!(Somebody, "some body", "somebody");
+create_closed_compound_linter!(Anybody, "any body", "anybody");
+create_closed_compound_linter!(Nothing, "no thing", "nothing");
+create_closed_compound_linter!(Anywhere, "any where", "anywhere");
+create_closed_compound_linter!(Instead, "in stead", "instead");
+create_closed_compound_linter!(Somewhere, "some where", "somewhere");
+create_closed_compound_linter!(Middleware, "middle ware", "middleware");
+create_closed_compound_linter!(Into, "in to", "into");
```

Comment:
Both are legit. As Grammarly apparently says: “Into” is a preposition that shows what something is within or inside. As separate words, in and to sometimes simply wind up next to each other.